### DISCUSSION TITLE
Implement systematic uncertainties

### DISCUSSION THREAD

**Discussion by maxgalli** [zfit/zfit#356] [Q&A]
Hi, I'm trying to implement the exercise described [here](https://cms-analysis.github.io/HiggsAnalysis-CombinedLimit/part5/longexercise.html) using zfit and hepstats. 
Combine is the tool which is used for most of the Higgs related analyses in CMS and it is based on the concept of datacard to provide the information necessary for the fit. A description of the main fields provided in a simple 1 bin counting experiment is shown in the link.

Here is my question: how can the systematic uncertainties be implemented in zfit? Quoting from the link:

> The remaining lines describe sources of systematic uncertainty. Each line gives the name of the uncertainty, (which will become the name of the nuisance parameter inside our RooFit model), the type of uncertainty ("lnN" = log-normal normalisation uncertainty) and the effect on each process in each channel. E.g. a 20% uncertainty on the yield is written as 1.20.

I was thinking, for this simple case, that adding a ```zfit.constraint.GaussianConstraint``` for each systematic might be the solution, but I'm not sure.

For completeness, a more detailed description of the types of systematic uncertainties is provided [here](https://cms-analysis.github.io/HiggsAnalysis-CombinedLimit/part2/settinguptheanalysis.html#a-simple-counting-experiment).

Thanks,

Massimiliano

**Comment by jonas-eschle** [comment]
Hi Massimiliano, that is a really interesting thing to do. (I am just traveling so I'll gladly have a more detailed look in two weeks from now). I think this is the right way to go, indeed. You can use Gaussian constraints or arbitrary constraints, either created by yourself or by using the base class for TFP distributions and wrapping any available distribution [here](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions?version=nightly).

Please just come forward with any question, I don't see any any problem with doing it but I am interested to see how it works myself. And, be picky with the zfit and hepstats code, it may be that there are things which are missing in the library (but they should, hopefully, be well readable.
